crypto/keys_test.go

```diff
@@ -1,209 +1,41 @@
-
-		reconstructedPriv, err := BytesToPrivKey(privKeyBytes, P224)
-		assert.NoError(tt, err)
-		assert.NotEmpty(tt, reconstructedPriv)
-		assert.EqualValues(tt, priv, reconstructedPriv)
-	})
-
-	t.Run("P256", func(tt *testing.T) {
-		pub, priv, err := GenerateKeyByKeyType(P256)
-		assert.NoError(tt, err)
-		assert.NotEmpty(tt, pub)
-		assert.NotEmpty(tt, priv)
-
-		pubKeyBytes, err := PubKeyToBytes(pub)
-		assert.NoError(tt, err)
-		assert.NotEmpty(tt, pubKeyBytes)
-
-		reconstructedPub, err := BytesToPubKey(pubKeyBytes, P256)
-		assert.NoError(tt, err)
-		assert.NotEmpty(tt, reconstructedPub)
-		assert.EqualValues(tt, pub, reconstructedPub)
-
-		privKeyBytes, err := PrivKeyToBytes(priv)
-		assert.NoError(tt, err)
-		assert.NotEmpty(tt, privKeyBytes)
-
-		reconstructedPriv, err := BytesToPrivKey(privKeyBytes, P256)
-		assert.NoError(tt, err)
-		assert.NotEmpty(tt, reconstructedPriv)
-		assert.EqualValues(tt, priv, reconstructedPriv)
-	})
-
-	t.Run("P384", func(tt *testing.T) {
-		pub, priv, err := GenerateKeyByKeyType(P384)
-		assert.NoError(tt, err)
-		assert.NotEmpty(tt, pub)
-		assert.NotEmpty(tt, priv)
-
-		pubKeyBytes, err := PubKeyToBytes(pub)
-		assert.NoError(tt, err)
-		assert.NotEmpty(tt, pubKeyBytes)
-
-		reconstructedPub, err := BytesToPubKey(pubKeyBytes, P384)
-		assert.NoError(tt, err)
-		assert.NotEmpty(tt, reconstructedPub)
-		assert.EqualValues(tt, pub, reconstructedPub)
-
-		privKeyBytes, err := PrivKeyToBytes(priv)
-		assert.NoError(tt, err)
-		assert.NotEmpty(tt, privKeyBytes)
-
-		reconstructedPriv, err := BytesToPrivKey(privKeyBytes, P384)
-		assert.NoError(tt, err)
-		assert.NotEmpty(tt, reconstructedPriv)
-		assert.EqualValues(tt, priv, reconstructedPriv)
-	})
-
-	t.Run("P521", func(tt *testing.T) {
-		pub, priv, err := GenerateKeyByKeyType(P521)
-		assert.NoError(tt, err)
-		assert.NotEmpty(tt, pub)
-		assert.NotEmpty(tt, priv)
-
-		pubKeyBytes, err := PubKeyToBytes(pub)
-		assert.NoError(tt, err)
-		assert.NotEmpty(tt, pubKeyBytes)
-
-		reconstructedPub, err := BytesToPubKey(pubKeyBytes, P521)
-		assert.NoError(tt, err)
-		assert.NotEmpty(tt, reconstructedPub)
-		assert.EqualValues(tt, pub, reconstructedPub)
-
-		privKeyBytes, err := PrivKeyToBytes(priv)
-		assert.NoError(tt, err)
-		assert.NotEmpty(tt, privKeyBytes)
-
-		reconstructedPriv, err := BytesToPrivKey(privKeyBytes, P521)
-		assert.NoError(tt, err)
-		assert.NotEmpty(tt, reconstructedPriv)
-		assert.EqualValues(tt, priv, reconstructedPriv)
-	})
+	for _, keyType := range GetSupportedKeyTypes() {
+		t.Run(string(keyType), func(tt *testing.T) {
+			pub, priv, err := GenerateKeyByKeyType(keyType)
+			validateKeyGeneration(tt, pub, priv, keyType, err)
+		})
+	}
+}
 
-	t.Run("RSA", func(tt *testing.T) {
-		pub, priv, err := GenerateKeyByKeyType(RSA)
-		assert.NoError(tt, err)
-		assert.NotEmpty(tt, pub)
-		assert.NotEmpty(tt, priv)
+func validateKeyGeneration(tt *testing.T, pub crypto.PublicKey, priv crypto.PrivateKey, keyType KeyType, err error) {
```

Comment:
I'm not sure I see the benefit in having this as a separate function.

Perhaps just embedding it in the above is sufficient?

Reply:
Yup embedding is a good call